src/Database/Commands/DeleteByQueryCommand.ts

```diff
@@ -26,7 +27,7 @@ export class DeleteByQueryCommand extends QueryBasedCommand  implements IJSonSer
   public setResponse(response: IResponse): IRavenResponse | IRavenResponse[] | void {
     const result: IRavenResponse = <IRavenResponse>super.setResponse(response);
 
-    if (!response.body) {
+    if (!response.body && StatusCodes.isError(response.statusCode)) {
       throw new IndexDoesNotExistException('Could not find index');
```

Comment:
Index does not exist only if you get 404 Not found. Other status codes, even if 4xx or 5xx may not mean that. Also please remove the `!response.body` check from the condition here. Let's do the following
1. if 404 => index does not exist
2. StatusCodes.isError => throw more generic exception (ErrorResponseException for example)